WordPress/wordpress-playground
# Add cache version number

## Motivation for the change, related issues

To implement [offline support](https://github.com/WordPress/wordpress-playground/pull/1535) we need to flush the cache every time Playground is updated.

This PR introduces a `cacheVersion` to the web service worker which will be used in https://github.com/WordPress/wordpress-playground/pull/1535 for cache busting. 

In this PR it doesn't do anything, but I wanted to add it as a separate PR to keep https://github.com/WordPress/wordpress-playground/pull/1535 reasonably large. 

## Implementation details

This PR moves various Vite extensions into a shared `vite-extensions` folder to avoid code duplication.

It adds a `buildVersionPlugin` Vite extension to create build versions in different projects (website and remote).

The new build version is created in the _Playground remote_ project and sent to _PHP-wasm Web service worker_ in `initializeServiceWorker` options. 

I had issues with implementing the build version directly in  _PHP-wasm Web service worker_, so I ended up working around the issue to unblock offline support. 
I recommend we move forward with this approach and change it if needed later.

A benefit of setting the cache version in Remote is that Playground can control when the cache is busted. 

## Testing Instructions (or ideally a Blueprint)

- Checkout this branch
- add the bellow code [after this line](https://github.com/WordPress/wordpress-playground/blob/3023e5b67767c586a5d160f7ab95519c84f77992/packages/php-wasm/web-service-worker/src/initialize-service-worker.ts#L14)
```
console.log('Service worker initialized', config.cacheVersion);
```
- Delete the service worker in dev tools
- Reload Playground 
- See the version number in the browser console
